Pcap++/src/PcapFilter.cpp

```diff
@@ -221,24 +216,54 @@ void IPFilter::convertToIPAddressWithLen(std::string& ipAddrmodified) const
 		addrAsInt &= mask;
 		ipAddrmodified = IPv4Address(addrAsInt).toString();
 	}
+	else if (ipAddr.getType() == IPAddress::IPv6AddressType)
+	{
+		std::array<uint8_t, 16> addrAsBytes{};
+		{
+			const uint8_t* bytesArrInternal = ipAddr.getIPv6().toBytes();
+			std::copy(bytesArrInternal, bytesArrInternal + 16, addrAsBytes.begin());
+		}
+
+		std::array<uint8_t, 16> mask{};
+		const int fullyMaskedBytes = m_Len / 8;
+		const int partialMaskBits = m_Len % 8;
+		for (size_t i = 0; i < static_cast<size_t>(fullyMaskedBytes) && i < mask.size(); i++)
+		{
+			mask[i] = static_cast<uint8_t>(-1);
+		}
+		if (static_cast<size_t>(fullyMaskedBytes) < mask.size())
+		{
+			mask[fullyMaskedBytes] = static_cast<uint8_t>(-1) << (8 - partialMaskBits);
+		}
+		for (size_t i = 0; i < addrAsBytes.size(); i++)
+		{
+			addrAsBytes[i] &= mask[i];
+		}
+
+		ipAddrmodified = IPv6Address(addrAsBytes.data()).toString();
+	}
+	else
+	{
+		// Realistically the code flow should never get to this block.
+		throw std::logic_error("Unsupported IP address version.");
+	}
 }
 
 void IPFilter::parseToString(std::string& result)
 {
 	std::string dir;
-	std::string ipAddr = m_Address;
+	std::string ipAddr = m_Address.toString();
 	std::string mask = m_IPv4Mask;
+	std::string ipProto = m_Address.isIPv6() ? "ip6" : "ip";
 	convertToIPAddressWithMask(ipAddr, mask);
 	convertToIPAddressWithLen(ipAddr);
 	parseDirection(dir);
-	result = "ip and " + dir + " net " + ipAddr;
+	result = ipProto + " and " + dir + " net " + ipAddr;
 	if (m_IPv4Mask != "")
 		result += " mask " + mask;
 	else if (m_Len > 0)
 	{
-		std::ostringstream stream;
-		stream << m_Len;
-		result += '/' + stream.str();
+		result += '/' + std::to_string(m_Len);
```

Comment:
`'/' + std::to_string(m_Len);` creates one more string, we can reuse `result` :
```
result += '/';
result += std::to_string(m_Len);
```
but I am not sure if this really matters the performance.

Reply:
Implemented in 120fb7d. Added a reserve call to result before string assignment and removed extra string constructions made by chaining `+` operators.